### ISSUE TITLE
cwl.inputs.json has wrong paths to input files

### ISSUE THREAD

**Issue by danbills** [broadinstitute/cromwell#3639]
When running bcbio somatic-giab-mix, there is a step that is using an initialworkdirrequirement:

```yaml
requirements:
- class: InlineJavascriptRequirement
- class: InitialWorkDirRequirement
  listing:
  - entry: $(JSON.stringify(inputs))
    entryname: cwl.inputs.json
```

This is generating a cwl.inputs.json as follows, pointing to either the base directory or the execution directory within docker (I'm not sure which this is):
```json
  "files": [
    [
      {
        "nameext": ".gz",
        "location": "/cromwell-executions/na12878-na24385-somatic/NA12878-NA24385-tumor_R1.fq.gz",
        "path": "/cromwell-executions/na12878-na24385-somatic/NA12878-NA24385-tumor_R1.fq.gz",
        "size": 13990775194,
        "dirname": "/cromwell-executions/na12878-na24385-somatic",
        "secondaryFiles": [],
        "basename": "NA12878-NA24385-tumor_R1.fq.gz",
        "class": "File",
        "nameroot": "NA12878-NA24385-tumor_R1.fq"
      },
      {
...
```

However, the inputs are *not* in the base directory, they're in the inputs directory:
![image](https://user-images.githubusercontent.com/165320/40139972-95c066f4-591f-11e8-8429-d39879328bc5.png)

this is not a blocker for running on SLURM (which does not use docker) but it is a blocker for anyone trying to run this workflow in a docker container

**Comment by geoffjentry** [comment]
@Horneth since you're actively working on this workflow, can you confirm if this is/isn't fixed, and close if so?

**Comment by illusional** [comment]
Random google-r checking in, I had some similar confusion with CWL and InitialWorkDirRequirements and although it's not directly related to this issue as I wasn't replicating with docker, I wanted to comment in case anyone else finds themselves here, especially within Cromwell.

My confusion can because I had a CommandLineTool that would only generate its files where the inputs are. Basically I needed to localise the file to the execution directory, and then reference it in a glob expression on output.

TL;DR, put the InitialWorkDirRequirement file expression in a `Dirent` rather than on the requirement.

 By _carefully_ re-reading the [`InitialWorkDirRequirement`](https://www.commonwl.org/v1.0/Workflow.html#InitialWorkDirRequirement).[`Dirent`](https://www.commonwl.org/v1.0/Workflow.html#InitialWorkDirRequirement) documentation, we find:

#### InitialWorkDirRequirement.listing
> May be an expression. If so, the expression return value must validate as {type: array, items: [File, Directory]}

This is what I initially did, which looks like:
```yaml
requirements:
  InitialWorkDirRequirement:
    listing: $([inputs.inputFile])
```
but my output expression was never localised to this new directory. Let's look at Dirent:

#### Dirent.entry
> If the value is an expression that evaluates to a File object, this indicates the referenced file should be added to the designated output directory prior to executing the tool.

Okay cool, this is basically what I want, and paired with:

#### Dirent.entryname
> The name of the file or subdirectory to create in the output directory. If entry is a File or Directory, the entryname field overrides the value of basename of the File or Directory object. Optional.

This is exactly what I want. I need to provide the expression for Dirent.entry, and exclude entryname, and Cromwell successfully localises this. Hence putting this together so that a tabix CommandLineTool can generate the file in the following CommandLineTool:

```cwl
class: CommandLineTool
baseCommand: tabix
requirements:
- class: InlineJavascriptRequirement
- class: InitialWorkDirRequirement
  listing:
  - entry: $(inputs.inputFile)
inputs:
    inputFile: File
```

I know this isn't the best place for this, but this is where I found myself when I was having struggles localising files in Cromwell. Feel free to delete if so.
